flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/AbstractDispatcherResourceManagerComponentFactory.java

```diff
@@ -260,7 +260,7 @@ public AbstractDispatcherResourceManagerComponentFactory(
 
 	protected String getHostname(RpcService rpcService) {
 		final String rpcServiceAddress = rpcService.getAddress();
-		return rpcServiceAddress == "" ? "localhost" : rpcServiceAddress;
+		return "".equals(rpcServiceAddress) ? "localhost" : rpcServiceAddress;
```

Comment:
Good catch. I think it would be even better to use `rpcServiceAddress.isEmpty()`.

Reply:
Yes, `RpcService.getAddress` has guaranteed that the return value cannot be `null`, hence we can call `isEmpty()` safely.